### ISSUE TITLE
Spring RequestParam annotation always generates a query type parameter

### ISSUE THREAD

**Issue by trevoreckersley** [stoicflame/enunciate#812]
We have a project where some of our endpoints use @RequestParam to capture a query parameter and other endpoints where this captures a form parameter. In both cases, Enunciate documents the parameter as a query parameter. We noticed that Enunciate has a `@ResourceMethodSignature`, but when we tried using it force Enunciate to treat some of those fields as form parameters, it seems to be ignored. Shortened example: `@ResourceMethodSignature(formParams = {@FormParam("name")})`

Should `@ResourceMethodSignature` be overriding the default docmentation handling of `@RequestParam`, or is there some other way of getting the desired behavior?

**Comment by stoicflame** [comment]
Okay. I'll look into why it's ignored and I'll consider adding some kind of mechanism for distinguishing a form `@RequestParam` and a query `@RequestParam`.

**Comment by epabst** [comment]
An alternative would be to simply not indicate in the documentation that it is a query param or a form param for RestController endpoints.

**Comment by stoicflame** [comment]
At da11136, I added some logic to assume that if the request method defines a url-encoded HTTP POST, then `@RequestParam` means "form". Hopefully that's an adequate distinction. If not, we'll have to add yet another annotation to make the distinction explicit.

I also looked into the non-effect of `@ResourceMethodSignature`. It turns out it only applies to JAX-RS APIs. Spring's API's are too "squishy" to deterministically map `@ResourceMethodSignature` to a request mapping.
